soh/src/code/z_en_item00.c

```diff
@@ -509,8 +510,13 @@ void EnItem00_Init(Actor* thisx, GlobalContext* globalCtx) {
     }
 
     if ((gSaveContext.n64ddFlag || getItemId != GI_NONE) && !Actor_HasParent(&this->actor, globalCtx)) {
-        getItemId = Randomizer_GetRandomizedItemId(getItemId, this->actor.id, this->ogParams, globalCtx->sceneNum);
-        func_8002F554(&this->actor, globalCtx, getItemId);
+        getItem = Randomizer_GetRandomizedItem(getItemId, this->actor.id, this->ogParams, globalCtx->sceneNum);
+        getItemId = getItem.getItemId;
+        if (!gSaveContext.n64ddFlag) {
+            func_8002F554(&this->actor, globalCtx, getItemId);
+        } else {
```

Comment:
should we have this block here? seems it might make sense to move it up a couple lines outside the `if ((gSaveContext.n64ddFlag || getItemId != GI_NONE) && !Actor_HasParent(&this->actor, globalCtx)) {` block, or maybe rework it to be something like

```c
if (!Actor_HasParent(&this->actor, globalCtx)) {
    if (!gSaveContext.n64ddFlag) {
        if (getItemId != GI_NONE) {
            func_8002F554(&this->actor, globalCtx, getItemId);
        }
    } else {
        GiveItemEntryFromActorWithFixedRange(&this->actor, globalCtx, Randomizer_GetRandomizedItem(getItemId, this->actor.id, this->ogParams, globalCtx->sceneNum));
    }
} 
```

mostly just trying to think of ways to avoid messing with the local `getItemId` var when rando'd because we're not using it (i know the nested ifs in the example code there aren't the best

Reply:
Hmm, I'll revisit this and see if we can clean it up. Worth noting is that, with few exceptions, we actually set both `getItemEntry` and `getItemId` In our `GiveItemEntry...` functions. This is to prevent odd scenarios with things like bugs, fish, and fairies, that actually set `GI_MAX` as the player's `getItemId`. So if you're picking up an item nearby these, you don't have your `getItemEntry` being whatever's in the item and your `getItemId` being `GI_MAX`, since the item will supercede the `GI_MAX` from the "critters".

Reply:
Yeah I see what you mean now, yeah this can definitely be cleaned up.

Reply:
Resolved by e5bca727ac2bd799e6354b1052e188a29451f365

Reply:
Looks like with the latest commit we won't hit the else if we're rando'd and getItemId is GI_NONE. 

That is slightly different logically, is that OK?

Reply:
Ah good catch the semantics are a bit different on this one. If the getItemId were `GI_NONE` when not rando'd we would enter the else block when we probably shouldn't be doing anything at all. Was trying to remove a level of nesting from the if but in hindsight definitely not the correct play. I'll fix that.